src/In.ProjectEKA.HipService/Discovery/PatientDiscovery.cs

```diff
@@ -1,39 +1,62 @@
 namespace In.ProjectEKA.HipService.Discovery
 {
     using System;
+    using System.Collections.Generic;
     using System.Linq;
     using System.Threading.Tasks;
     using HipLibrary.Patient;
     using HipLibrary.Patient.Model;
+    using In.ProjectEKA.HipService.Link;
+    using In.ProjectEKA.HipService.Link.Model;
 
     public class PatientDiscovery : IDiscovery
     {
         private readonly Filter filter;
         private readonly IMatchingRepository matchingRepository;
         private readonly IDiscoveryRequestRepository discoveryRequestRepository;
+        private readonly ILinkPatientRepository linkPatientRepository;
 
         public PatientDiscovery(
             IMatchingRepository patientRepository,
-            IDiscoveryRequestRepository discoveryRequestRepository)
+            IDiscoveryRequestRepository discoveryRequestRepository,
+            ILinkPatientRepository linkPatientRepository)
         {
             matchingRepository = patientRepository;
             this.discoveryRequestRepository = discoveryRequestRepository;
+            this.linkPatientRepository = linkPatientRepository;
             filter = new Filter();
         }
 
         public async Task<Tuple<DiscoveryRepresentation, ErrorRepresentation>> PatientFor(DiscoveryRequest request)
         {
             var patients = await matchingRepository.Where(request);
             var (patient, error) = DiscoveryUseCase.DiscoverPatient(filter.Do(patients, request).AsQueryable());

```

Comment:
Should not we actually check whether patient's already linked. if linked, get patient detail by patient reference number and return only whatever you do right now?